napi.h

```diff
@@ -1236,66 +1236,86 @@ namespace Napi {
   class PropertyDescriptor {
```

Comment:
Did not review in depth, but thinking that instead of changing public API we should add new methods, deprecate the old ?  

While the commitment to stability for node-addon-api is lower than N-API itself, at this point I think it would be good to avoid forcing code changes when modules upgrade.